dotCMS/src/main/java/com/dotmarketing/portlets/contentlet/util/ContentletUtil.java

```diff
@@ -142,33 +164,37 @@ public static String sanitizeFileName(String fileName){
 			} else if(f instanceof CategoryField) {
 
 				List<Category> cats = null;
-				
 				try {
-
 					cats = APILocator.getCategoryAPI().getParents(contentlet, user, true);
 				} catch (Exception e) {
 					Logger.error(ContentletUtil.class, String.format("Unable to get the Categories for given contentlet with inode= %s", contentlet.getInode()));
 				}
 
 				if(cats!=null && !cats.isEmpty()) {
 					try {
-
 						final Category parentCategory        = APILocator.getCategoryAPI().find(f.values(), user, true);
 						final List<Category> childCategories = new ArrayList<>();
 
 						if(parentCategory != null) {
 							for (Category category : cats) {
-
 								if (APILocator.getCategoryAPI().isParent(category, parentCategory, user,true)) {
-
 									childCategories.add(category);
 								}
 							}
 						}
 
 						if (!childCategories.isEmpty()){
-							String catsStr = childCategories.stream().map(Category::getCategoryName).collect(Collectors.joining(", "));
-							m.put(f.variable(), catsStr);
+							Object categoriesValue;
+							if (allCategoriesInfo) {
+								categoriesValue = childCategories.stream()
+										.map(Category::getMap)
+										.collect(Collectors.toList());
+							} else {
+								categoriesValue = childCategories.stream()
+										.map(Category::getCategoryName)
+										.collect(Collectors.joining(", "));
+							}
+							m.put(f.variable(), categoriesValue);

```

Comment:
This is a great change and the code to include the categories is in the right place.